### PR TITLE
feat(sidebar): add support for customize sidebar

### PR DESCRIPTION
**Pull request by neoFelhz**
**What kind of change does this PR introduce?** (check one with "x")

- [ ] Bug fix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?** (check one with "x")

- [x] Yes
- [ ] No

____

**Description**

Add support for sidebar customize.
You can customize your sidebar icons in config file now.
(I remove the *I'm Feeling Lucky* item as well)

____

**Verification steps**

- You have to update your _config.yml in order to use this feature and avoid bug.

### CODE REVIEW

File: layout/_partial/sidebar-navigation.ejs [hunk 1]
```diff
@@ -14,44 +14,38 @@
...
-    <!--
-    <li class="dropdown">
-        <a href="" target="_self">
-            <i class="material-icons sidebar-material-icons">explore</i>
-             <%= __("sidebar.imlucky") %>
-        </a>
-    </li>
-    -->
+    <% if(theme.sidebar.homepage === true) { %>
+            <li id="sidebar-first-li">
```
Comment (pidupuis) [inline layout/_partial/sidebar-navigation.ejs]: - Please remove 4 spaces between lines 18, 19 and 20

File: layout/_partial/sidebar-navigation.ejs [hunk 2]
```diff
@@ -14,44 +14,38 @@
...
-        <a href="#" class="ripple-effect dropdown-toggle" data-toggle="dropdown">
-            <i class="material-icons sidebar-material-icons">inbox</i>
-             <%= __('sidebar.archive') %>
-            <b class="caret"></b>
-        </a>
-        <ul class="dropdown-menu">
+    <% if(theme.sidebar.archives === true) { %>
+        <li class="dropdown">
+            <a href="#" class="ripple-effect dropdown-toggle" data-toggle="dropdown">
+                <i class="material-icons sidebar-material-icons"><%= theme.sidebar.archive.icon %></i>
```
Comment (pidupuis) [inline layout/_partial/sidebar-navigation.ejs]: - It should be `theme.sidebar.archives.icon` (you forgot an 's')
- Please surround this line with the following condition: `if(theme.sidebar.archives.icon){`

File: layout/_partial/sidebar-navigation.ejs [hunk 3]
```diff
@@ -14,44 +14,38 @@
...
+            </ul>
+        </li>
+    <% } %>
 
     <!-- Categories  -->
-    <% if(theme.sidebar_categories === true) { %>
+    <% if(theme.sidebar.categories === true) { %>
         <li class="dropdown">
             <a href="#" class="ripple-effect dropdown-toggle" data-toggle="dropdown">
                 <i class="material-icons sidebar-material-icons">chrome_reader_mode</i>
```
Comment (pidupuis) [inline layout/_partial/sidebar-navigation.ejs]: - You seem to have forgotten to use the icon from the configuration like `<%= theme.sidebar.categories.icon %>`
- And then surround this line with the following condition: `if(theme.sidebar.categories.icon){`

File: layout/_partial/sidebar-navigation.ejs [hunk 4]
```diff
@@ -2,10 +2,10 @@
     <!-- User dropdown  -->
     <li class="dropdown">
         <ul id="settings-dropdown" class="dropdown-menu">
-            <% for (var i in theme.dropdown) { %>
+            <% for (var i in theme.sidebar.dropdown) { %>
                 <li>
-                    <a href="<%= theme.dropdown[i].link %>" target="_blank" title="<%= i %>">
-                        <i class="material-icons sidebar-material-icons sidebar-indent-left1pc-element"><%= theme.dropdown[i].icon %></i>
+                    <a href="<%= theme.sidebar.dropdown[i].link %>" target="_blank" title="<%= i %>">
+                        <i class="material-icons sidebar-material-icons sidebar-indent-left1pc-element"><%= theme.sidebar.dropdown[i].icon %></i>
```
Comment (pidupuis) [inline layout/_partial/sidebar-navigation.ejs]: - Please surround this line with the following condition: `if(theme.sidebar.dropdown[i].icon){`

**Comment by neoFelhz** [comment]
@pidupuis OK，I will fix it.

**Comment by neoFelhz** [comment]
@pidupuis Now I push a new commit to fix them,please review them again(And sorry for my carelessness)

**Pull request merged**